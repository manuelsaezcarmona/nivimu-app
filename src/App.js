import { Table } from 'antd';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './App.css';
import { getAllUsers } from './redux/user.actioncreator';
import { columns } from './services/columns';

function App() {
  // eslint-disable-next-line arrow-parens
  const { users } = useSelector(store => store.user);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAllUsers());
  }, [dispatch]);

  return (
    <div>
      <h2>Test instructions</h2>
      <ol>
        <li>
          <span>
            Create table comp from
            <a target="_blank" rel="noreferrer" href="https://ant.design/">
              Ant Design
            </a>
            with 3 columns: Name, Email, City.
          </span>
        </li>
        <li>
          <span>Load data for table from: </span>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://jsonplaceholder.typicode.com/users"
          >
            https://jsonplaceholder.typicode.com/users
          </a>
          .
        </li>
        <li>
          <span>
            Use
            <strong>
              <a href="https://redux.js.org/" target="_blank" rel="noreferrer">
                Redux
              </a>
            </strong>
            or alternative.
          </span>
        </li>
        <li>
          <span>
            Add <strong>filter</strong> and <strong>sorter</strong>
            functionality to table
          </span>
        </li>
        <li>
          <span>
            Add new component to <strong>display always first row data</strong>{' '}
            of table
          </span>
        </li>
      </ol>
      <i>Estimate time: 1h</i>
      <Table
        dataSource={users}
        columns={columns}
        pagination={{ pageSize: 5 }}
        scroll={{ y: 240 }}
      />
      ;
    </div>
  );
}

export default App;
