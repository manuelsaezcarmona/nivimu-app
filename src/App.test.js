import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Table } from 'antd';
import configureStore from './redux/store';

import App from './App';

jest.mock('antd', () => ({
  Table: () => {
    const Table = 'Table';
    return <Table />;
  }
}));

test('renders App ', () => {
  const page = render(
    <Provider store={configureStore()}>
      <App />
    </Provider>
  );
  const ulElement = screen.getByText(/add new component to of table/i);
  const tableElement = screen.getByRole('table');
  expect(ulElement).toBeInTheDocument();
  expect(tableElement).toBeInTheDocument();
  expect(page).toMatchSnapshot();
});
