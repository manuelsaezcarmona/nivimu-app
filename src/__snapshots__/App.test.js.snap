// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Test instructions
        </h2>
        <ol>
          <li>
            <span>
              Create table comp from
              <a
                href="https://ant.design/"
                rel="noreferrer"
                target="_blank"
              >
                Ant Design
              </a>
              with 3 columns: Name, Email, City.
            </span>
          </li>
          <li>
            <span>
              Load data for table from: 
            </span>
            <a
              href="https://jsonplaceholder.typicode.com/users"
              rel="noreferrer"
              target="_blank"
            >
              https://jsonplaceholder.typicode.com/users
            </a>
            .
          </li>
          <li>
            <span>
              Use
              <strong>
                <a
                  href="https://redux.js.org/"
                  rel="noreferrer"
                  target="_blank"
                >
                  Redux
                </a>
              </strong>
              or alternative.
            </span>
          </li>
          <li>
            <span>
              Add 
              <strong>
                filter
              </strong>
               and 
              <strong>
                sorter
              </strong>
              functionality to table
            </span>
          </li>
          <li>
            <span>
              Add new component to 
              <strong>
                display always first row data
              </strong>
               
              of table
            </span>
          </li>
        </ol>
        <i>
          Estimate time: 1h
        </i>
        <table />
        ;
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Test instructions
      </h2>
      <ol>
        <li>
          <span>
            Create table comp from
            <a
              href="https://ant.design/"
              rel="noreferrer"
              target="_blank"
            >
              Ant Design
            </a>
            with 3 columns: Name, Email, City.
          </span>
        </li>
        <li>
          <span>
            Load data for table from: 
          </span>
          <a
            href="https://jsonplaceholder.typicode.com/users"
            rel="noreferrer"
            target="_blank"
          >
            https://jsonplaceholder.typicode.com/users
          </a>
          .
        </li>
        <li>
          <span>
            Use
            <strong>
              <a
                href="https://redux.js.org/"
                rel="noreferrer"
                target="_blank"
              >
                Redux
              </a>
            </strong>
            or alternative.
          </span>
        </li>
        <li>
          <span>
            Add 
            <strong>
              filter
            </strong>
             and 
            <strong>
              sorter
            </strong>
            functionality to table
          </span>
        </li>
        <li>
          <span>
            Add new component to 
            <strong>
              display always first row data
            </strong>
             
            of table
          </span>
        </li>
      </ol>
      <i>
        Estimate time: 1h
      </i>
      <table />
      ;
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
